api/naming/naming.go

```diff
@@ -27,3 +28,17 @@ func Task(t *api.Task) string {
 }
 
 // TODO(stevvooe): Consolidate "Hostname" style validation here.
+
+// Runtime returns the runtime name from a given spec.
+func Runtime(t api.TaskSpec) string {
+	switch r := t.GetRuntime().(type) {
+	case *api.TaskSpec_Attachment:
+		return "attachment"
+	case *api.TaskSpec_Container:
+		return "container"
+	case *api.TaskSpec_Generic:
+		return strings.ToLower(r.Generic.Kind)
+	default:
+		return "none"
```

Comment:
I don't think it should panic. But I think returning an error is the right thing to do. Returning `"none"` here means that passing `"none"` as the filter parameter would return unrecognized runtimes, which would be a weird undocumented quirk in the API.